source/extensions/filters/http/basic_auth/basic_auth_filter.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include "envoy/stats/stats_macros.h"
+
+#include "source/common/common/logger.h"
+#include "source/extensions/filters/http/common/pass_through_filter.h"
+
+#include "absl/container/flat_hash_map.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace BasicAuth {
+
+/**
+ * All Basic Auth filter stats. @see stats_macros.h
+ */
+#define ALL_BASIC_AUTH_STATS(COUNTER)                                                              \
+  COUNTER(allowed)                                                                                 \
+  COUNTER(denied)
+
+/**
+ * Struct definition for Basic Auth stats. @see stats_macros.h
+ */
+struct BasicAuthStats {
+  ALL_BASIC_AUTH_STATS(GENERATE_COUNTER_STRUCT)
+};
+
+/**
+ * Struct definition for username password pairs.
+ */
+struct User {
+  // the user name
+  std::string name;
+  // the hashed password,  see https://httpd.apache.org/docs/2.4/misc/password_encryptions.html
+  std::string hash;
+};
+
+using UserMap = absl::flat_hash_map<std::string, User>; // username, User
```

Comment:
Couldn't we use normal asbl::flat_hash_map<std::string, std::string> directly here?